src/VCR/Storage/AbstractStorage.php

```diff
@@ -50,12 +51,23 @@
     /**
      * Creates a new file store.
      *
+     * If the cassetteName contains PATH_SEPARATORs, subfolders of the
+     * cassettePath are autocreated when not existing.
+     *
      * @param string $cassettePath Path to the cassette directory.
-     * @param string $cassetteName Path to a file, will be created if not existing.
+     * @param string $cassetteName Path to the cassette file, relative to the path.
      */
     public function __construct($cassettePath, $cassetteName, $defaultContent = '[]')
     {
-        $file = $cassettePath . DIRECTORY_SEPARATOR . $cassetteName;
+        if (!is_dir($cassettePath)) {
+            throw new VCRException(sprintf('Cassette path "%s" is not existing or not a directory', $cassettePath), 0);
```

Comment:
Yes definitely it should be here too. I agree that it is useful in any case.

I guess you're referring to the `VCR` static class. In general I'm not a fan of singletons for various reasons. 

Some background on the reasons for using a singleton in this case: With the current implementation of PHP-VCR I don't see how two Videorecorder instances can exist at the same time. The function calls (`curl`/`SoapClient::*`/...) are overwritten globally and VCR is either intercepting or not. Being able to have more instances would lead to strange bugs. 

Thinking about this... it could be actually possible to whitelist/blacklist some parts of code that does HTTP requests which should be intercepted and use it with one videorecorder instance and another part of code with another. Currently this is not implemented and I'm not sure if it is even worth it ;-)